### ISSUE TITLE
Pace of releases over the past month

### ISSUE THREAD

**Issue by kevinlul** [motdotla/dotenv#616]
Hi, I've noticed that this package has seen five major version bumps over the span of three weeks. Is this a result of very active development and releasing early and often? I have some concerns about stability for a very widely used package and find it a bit odd that so many releases have happened.

**Comment by motdotla** [comment]
> Is this a result of very active development and releasing early and often

It is, yes. 

> I have some concerns about stability for a very widely used package and find it a bit odd that so many releases have happened

I completely understand. Dotenv is a very popular library with 25 million installs per week. It had not had many updates for quite a while, and it was overdue new features like:

* multiline parsing
* process overrides
* backtick support
* inline comments

The nature of maintaining such a highly relied on module combined with the nature of availability to work on it (other responsibilities) sometimes leads to these rapid back to back releases of major versions. We do our best to space them out but sometimes momentum catches hold and it's nice to release them more quickly.

**Comment by bashgeek** [comment]
@motdotla Was thinking the same thing today. While I appreciate the updates and work on it (please keep it coming! :-)), but do those really require a new major version? I didn't see any breaking or huge API changes with those updates, maybe minor update versions are enough for those in the future? Frequent new major versions will cause quite some trouble in the enterprise field, because it usually causes a whole process of investigation in compatibility and so on, because major version updates usually mean breaking changes and/or huge refactoring or similar. Having a couple of those new major versions in a few weeks can really ramp up workload here for no reasons :-)

Maybe something to consider in the future, thanks for your work!

**Comment by motdotla** [comment]
> Maybe something to consider in the future, thanks for your work!

Thank you 💛, and yes, thank you. I will keep this more at the forefront of my mind as well as those helping out. I've worked in some enterprise worlds and it is certainly not fun kicking off all that extra investigation.

**Comment by motdotla** [comment]
>  I didn't see any breaking or huge API changes with those updates, maybe minor update versions are enough for those in the future

To add some color here.. this repo is unique. The API is really the way the `.env` file is parsed. The API isn't really the method names. This is rare to modules. All of these major releases have slightly changed the way the `.env` file is parsed... which could cause breakage (far edge cases) for some users. Those far edge cases end up quite common though because it is so heavily used.

That's the decision process at least - for those wondering why the major releases.

TLDR: The API is actually the .env parsing engine - not the node module methods and arguments.
